server/coordinator/scheduler/manager/scheduler_manager.go

```diff
@@ -260,13 +279,51 @@ func (m *ManagerImpl) UpdateDeployMode(ctx context.Context, enable bool) error {
 	return nil
 }
 
-func (m *ManagerImpl) GetDeployMode(_ context.Context) (bool, error) {
+func (m *schedulerManagerImpl) GetDeployMode(_ context.Context) (bool, error) {
 	m.lock.RLock()
 	defer m.lock.RUnlock()
 
 	if m.topologyType != storage.TopologyTypeDynamic {
-		return false, errors.WithMessage(ErrInvalidTopologyType, "deploy mode could only get when topology type is dynamic")
+		return false, ErrInvalidTopologyType.WithCausef("deploy mode could only get when topology type is dynamic")
 	}
 
 	return m.deployMode, nil
 }
+
+func (m *schedulerManagerImpl) AddShardAffinityRule(ctx context.Context, rule scheduler.ShardAffinityRule) error {
+	var firstErr error
+	for _, scheduler := range m.registerSchedulers {
+		if err := scheduler.AddShardAffinityRule(ctx, rule); err != nil {
+			firstErr = err
+		}
+	}
+
+	return firstErr
```

Comment:
Add some comment to explain the meaning of error.

Reply:
I can't figure out how to give comments to an implicit error, could you provide some suggestions about such comments?

Reply:
The logic of error processing here is similar to errorgroup. I intercept the code in errorgroup.
```
// Go calls the given function in a new goroutine.
//
// The first call to return a non-nil error cancels the group; its error will be
// returned by Wait.
func (g *Group) Go(f func() error) {
	g.wg.Add(1)

	go func() {
		defer g.wg.Done()

		if err := f(); err != nil {
			g.errOnce.Do(func() {
				g.err = err
				if g.cancel != nil {
					g.cancel()
				}
			})
		}
	}()
}
```
Add a description of the error returned, like this:
```
// The first call to return a non-nil error cancels the group; its error will be
// returned by Wait.
```